app/src/main/java/com/example/campusguide/MapsActivity.kt

```diff
@@ -60,6 +62,24 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
         val hall = LatLng(45.497290, -73.578824)
         mMap.addMarker(MarkerOptions().position(hall).title("Hall Building"))
         mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(hall, 17.0f))
+
+        // Update switch campus button listener
+        val switchCampusToggle: ToggleButton = findViewById(R.id.switchCampusButton)
+        switchCampusToggle.setOnCheckedChangeListener{ _, isChecked ->
+            switchCampus(isChecked)
+        }
+    }
+    
+    private fun switchCampus(isChecked: Boolean) {
```

Comment:
I suggest we rename isChecked to isAtDowntownCampus to avoid having to explain in a comment what being checked means